src/functions/routers/types.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  FunctionContext,
+  FunctionDefinitionArgs,
+  FunctionParameters,
+  FunctionRuntimeParameters,
+} from "../types.ts";
+
+export type ActionHandler<Definition> = Definition extends
+  FunctionDefinitionArgs<infer I, infer O, infer RI, infer RO> ? {
+    (
+      context:
+        & Omit<FunctionContext<FunctionRuntimeParameters<I, RI>>, "event">
+        & ActionContext<FunctionRuntimeParameters<I, RI>>,
+    ): Promise<void> | void;
+  }
+  : never;
+
+export type ActionContext<InputParameters extends FunctionParameters> = {
+  body: BlockActionInvocationBody<InputParameters>;
+  action: BlockAction;
+};
+
+// TODO: all below stolen from deno-slack-runtime, need to centralize these somewhere
+export type BlockActionInvocationBody<
+  InputParameters extends FunctionParameters,
+> = {
+  type: "block_actions";
+  actions: BlockAction[];
+  bot_access_token?: string;
+  function_data?: FunctionData<InputParameters>;
```

Comment:
In the runtime I left this as optional for some backwards compatibility - as it's not always present until a full rollout happens. I believe we can treat it always present here, as the handler would never be invoked if it wasn't there. It would simplify how you pull out properties from here in your handler code a little, like getting the execution_id.

Reply:
I think I accidentally marked this comment resolved, but I agree with @selfcontained that it should be treated as always present here. We need it in order to complete the function via 
```ts
client.functions.completeSuccess({
    outputs: {},
    function_execution_id: body.function_data?.execution_id,
  });
```